lua/nvim-tree/async.lua

```diff
@@ -0,0 +1,180 @@
+---Idea taken from: https://github.com/ms-jpq/lua-async-await
```

Comment:
There's a lot of unused codepaths in here. Let's trim it down. If we do need functionality we can refer to the source.

It looks like:
`in_async`
`wrap`
`all`
interrupt/abort